contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -3,176 +3,312 @@ pragma solidity >=0.8.4;
 import "./PriceOracle.sol";
 import "./BaseRegistrarImplementation.sol";
 import "./StringUtils.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
 import "../resolvers/Resolver.sol";
+import "../registry/ReverseRegistrar.sol";
+import "../dnssec-oracle/BytesUtils.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@ensdomains/name-wrapper/interfaces/INameWrapper.sol";
+
+import "hardhat/console.sol";
+
+interface commitmentController {
+    function rentPrice(string memory, uint256) external;
+
+    function available(string memory) external;
+
+    function makeCommitment(
+        string memory,
+        address,
+        bytes32
+    ) external;
+
+    function commit(bytes32) external;
+
+    function register(
+        string calldata,
+        address,
+        uint256,
+        bytes32
+    ) external;
+
+    function renew(string calldata, uint256) external;
+}
 
 /**
  * @dev A registrar controller for registering and renewing names at fixed cost.
  */
 contract ETHRegistrarController is Ownable {
     using StringUtils for *;
-
-    uint constant public MIN_REGISTRATION_DURATION = 28 days;
-
-    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
-    bytes4 constant private COMMITMENT_CONTROLLER_ID = bytes4(
-        keccak256("rentPrice(string,uint256)") ^
-        keccak256("available(string)") ^
-        keccak256("makeCommitment(string,address,bytes32)") ^
-        keccak256("commit(bytes32)") ^
-        keccak256("register(string,address,uint256,bytes32)") ^
-        keccak256("renew(string,uint256)")
+    using BytesUtils for bytes;
+
+    uint256 public constant MIN_REGISTRATION_DURATION = 28 days;
+
+    bytes4 private constant INTERFACE_META_ID =
+        bytes4(keccak256("supportsInterface(bytes4)"));
+    bytes4 private constant COMMITMENT_CONTROLLER_ID =
+        type(commitmentController).interfaceId;
+    bytes4 private constant COMMITMENT_WITH_CONFIG_CONTROLLER_ID =
```

Comment:
I think this is the version we want an interface for; we're no longer supporting the version without config.